### PR TITLE
[extension_google_sign_in_as_googleapis_auth] Update example app.

### PR DESCRIPTION
**Pull request by ditman**
Update the example app of `package:extension_google_sign_in_as_googleapis_auth` by grabbing the latest version of the `example` app from `package:google_sign_in` and applying the changes required for this package to work.

The example app builds again on Android (at least on an emulator, it starts up!)

Fixes https://github.com/flutter/flutter/issues/89301

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by ditman** [comment]
@stuartmorgan I've copied a bunch of test-like stuff from the google_sign_in example from flutter/plugins, but I'm not sure if it's 100% appropriate (especially the `pbxproj`/iOS files)

I have verified that the app now builds on Android again, but it seems to lack the client ID to correctly use the Auth service.

I'm not sure if tests run correctly (I hope something happens in this run!)

**Comment by ditman** [comment]
(Force-pushed after fixing conflicts on a rebase with master)

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/build.gradle [hunk 1]
```diff
@@ -60,5 +61,7 @@ flutter {
 dependencies {
     implementation 'com.google.android.gms:play-services-auth:16.0.1'
     testImplementation'junit:junit:4.12'
-    testImplementation 'org.mockito:mockito-core:2.17.0'
+    androidTestImplementation 'androidx.test:runner:1.2.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/android/app/build.gradle:65]: We don't run Espresso tests in this repo currently, and I didn't see any in this PR, so I wouldn't expect us to need this.

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+package io.flutter.plugins;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface DartIntegrationTest {}
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java:14]: We're not running this file at all in this repo. Do we actually need native integration tests for this package?

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/googlesigninexample/FlutterActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import io.flutter.plugins.DartIntegrationTest;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+@DartIntegrationTest
+@RunWith(FlutterTestRunner.class)
+public class FlutterActivityTest {
+  @Rule
+  public ActivityTestRule<FlutterActivity> rule = new ActivityTestRule<>(FlutterActivity.class);
+}
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/googlesigninexample/FlutterActivityTest.java:19]: This is a plugin test; it doesn't make sense to include it.

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/googlesigninexample/GoogleSignInTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  @Test
+  public void googleSignInPluginIsAdded() {
+    final ActivityScenario<GoogleSignInTestActivity> scenario =
+        ActivityScenario.launch(GoogleSignInTestActivity.class);
+    scenario.onActivity(
+        activity -> {
+          assertTrue(activity.engine.getPlugins().has(GoogleSignInPlugin.class));
+        });
+  }
+}
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/googlesigninexample/GoogleSignInTest.java:23]: Same.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
I started doing a file-by-file review, but it seems like there's a lot here that's not relevant.

What about just nuking `example`, regenerating it from the template, and re-adding the Dart bits?

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java (hunk 2 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java:14]: This is just a result of doing a cp from the current implementation of the example app of google_sign_in into here. I thought you were looking at adding Android tests for this repo? I can remove these tests, no problem (there's probably the xcode equivalents somewhere under the ios dir?)

**Comment by ditman** [comment]
> What about just nuking example, regenerating it from the template, and re-adding the Dart bits?

Yep, this is better. I'll get this done.

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java (hunk 2 above)
Comment (stuartmorgan) [reply packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java]: We can run whatever tests in the repo that we believe are useful, but generally I wouldn't expect a non-plugin package to need to run native tests. Presumably this package is testable by shimming out the plugin it wraps, and doing Dart unit tests?

**Comment by ditman** [comment]
@stuartmorgan I completely nuked `example` and regenerated it:

```
flutter create googlesigninexample --org io.flutter.plugins
```

The only modifications I did were to `lib/main.dart`, the `pubspec.yaml` and ensured that android had the `gms` plugin enabled through gradle.

I added the plist file in the right spot on the ios code (inside the Runner), but I did it manually, so I'm sure it's missing some xcode magic :/

PTAL!

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java (hunk 2 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/androidTest/java/io/flutter/plugins/DartIntegrationTest.java]: Yes, there are a couple of unit-tests in the root test directory that are enough (the behavior is very simple).

I think the most valuable thing to do here would be to say: this example app passes the same integration tests as google_sign_in, in an attempt to prove they're functionally identical?

(The only change actually is that this example app uses `package:googleapis` and the standard app uses raw http requests with hand-coded authentication headers)

File: packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml [hunk 5]
```diff
@@ -4,8 +4,13 @@ publish_to: none
...
 environment:
   sdk: ">=2.12.0 <3.0.0"
-  flutter: ">=1.20.0"
 
+# Dependencies specify other packages that your package needs in order to work.
+# To automatically upgrade your package dependencies to the latest versions
+# consider running `flutter pub upgrade --major-versions`. Alternatively,
+# dependencies can be manually updated by changing the version numbers below to
+# the latest version available on pub.dev. To see which dependencies have newer
+# versions available, run `flutter pub outdated`.
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml:13]: Remove all the boilerplate comments please.

File: packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml [hunk 6]
```diff
@@ -4,8 +4,13 @@ publish_to: none
 
 environment:
   sdk: ">=2.12.0 <3.0.0"
-  flutter: ">=1.20.0"
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml:7]: Let's re-add this.

File: packages/extension_google_sign_in_as_googleapis_auth/example/test/widget_test.dart [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    await tester.pumpWidget(
+      const MaterialApp(
+        title: 'Google Sign In',
+        home: SignInDemo(),
+      ),
+    );
+
+    expect(find.byKey(signInButtonKey), findsOneWidget);
+  });
+}
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/test/widget_test.dart:29]: Remove this file

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits.

Some of the added churn here is because you didn't use `--ios-language=objc --android-language=java`, but for an example in the packages repo I think it probably doesn't matter. (In plugins we want the example runners to use the same language as everything else, at least for now.)

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/main/kotlin/io/flutter/plugins/googlesigninexample/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package io.flutter.plugins.googlesigninexample
```
Comment (stuartmorgan) [inline packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/main/kotlin/io/flutter/plugins/googlesigninexample/MainActivity.kt:1]: Oops, needs a license. (Tool fix PR incoming.)

**Comment by ditman** [comment]
> you didn't use --ios-language=objc --android-language=java, but for an example in the packages repo I think it probably doesn't matter.

This is a good point, I'll get it done before merging, see if it simplifies things :)

File: packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/main/kotlin/io/flutter/plugins/googlesigninexample/MainActivity.kt (hunk 8 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/example/android/app/src/main/kotlin/io/flutter/plugins/googlesigninexample/MainActivity.kt:1]: Good catch! I'll regenerate this with `--android-language=java` so the kt file will go away

File: packages/extension_google_sign_in_as_googleapis_auth/example/test/widget_test.dart (hunk 7 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/example/test/widget_test.dart:29]: I'll also clean up the Key I added for that locator :D

File: packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml (hunk 6 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml:7]: Will do, I thought 1.20 was already "very old", but it won't hurt to have this check here.

File: packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml (hunk 5 above)
Comment (ditman) [reply packages/extension_google_sign_in_as_googleapis_auth/example/pubspec.yaml:13]: You got it :D

**Comment by ditman** [comment]
Ready to go!

**Pull request merged**